lib/srv/desktop/rdp/rdpclient/src/lib.rs

```diff
@@ -261,13 +263,50 @@ fn connect_rdp_inner(
         Ok(())
     });
 
+    let tdp_sd_info_request = Box::new(move |req: SharedDirectoryInfoRequest| -> RdpResult<()> {
+        debug!("sending: {:?}", req);
+        // Create C compatible string from req.path
+        match CString::new(req.path.clone()) {
+            Ok(c_string) => {
+                unsafe {
+                    // Convert C compatible string into raw pointer for passing into the CGO struct
+                    let c_string = c_string.into_raw();
+                    let err = tdp_sd_info_request(
+                        go_ref,
+                        &mut CGOSharedDirectoryInfoRequest {
```

Comment:
Does this need to be a mutable ref?

Reply:
It did when the function's signature had `req: *mut CGOSharedDirectoryInfoRequest` but now I've changed that to `req: *const CGOSharedDirectoryInfoRequest`

Reply:
Nevermind, that doesn't work. It seems as though `*mut SomeType` in Rust is equivalent to `struct SomeType *` in C. If I use `*const SomeType` in Rust, the compiler complains about a conflicting type coming from `librdprs.h`.

Reply:
Yup, *const T and &T do not work across the boundary. This is a restriction of cbindgen because there isn't a good way to enforce and make this not hilariously easy to misuse causing heisenbugs.